cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1696,7 +1854,23 @@ CacheAllocator<CacheTrait>::findEviction(TierId tid, PoolId pid, ClassId cid) {
       XDCHECK(candidate->getKey() == evictedToNext->getKey());
 
       (*stats_.numWritebacks)[tid][pid][cid].inc();
-      wakeUpWaiters(*candidate, std::move(evictedToNext));
+      if (chainedItem) {
+          XDCHECK(toRecycleParent->isMoving());
+          XDCHECK_EQ(evictedToNext->getRefCount(),2u);
+          evictedToNext.reset(); //destroy new chained item hdl
+          auto ref = toRecycleParent->unmarkMoving();
+          auto parentHandle = acquire(toRecycleParent);
+          if (parentHandle) {
+            wakeUpWaiters(*toRecycleParent,std::move(parentHandle));
+          } else if (UNLIKELY(ref == 0)) {
```

Comment:
Is it possible to have ``parentHandle == NULL` and `ref != 0`? I mean, is there any `else` we need to handle? If no, perhaps let's add an assert?